packages/client/src/game-engine/core/card/cardComponent/cardRect.ts

```diff
@@ -0,0 +1,221 @@
+import { CardTypeEmum } from '../../types/card'
+import { ICoordinates, OrientationEnum, PositionEnum } from '../../types/card'
+import { fillRect } from '../../../utils/fillRect'
+import { strokeRect } from '../../../utils/strokeRect'
+import { CanvasElement } from '../../canvas/canvasElement'
+import { ThemeConfig } from '../../../config/monopolyConfig'
+
+export interface ICardRect {
+  index: number
+  canvasSize: number
+  ctx: CanvasRenderingContext2D
+  type: CardTypeEmum
+}
+
+export abstract class CardRect extends CanvasElement {
+  static readonly CONST = {
+    //индексы карт
+    INDEX_CORNER: 10,
+    INDEX_TOP_LEFT: 0,
+    INDEX_TOP_RIGHT: 10,
+    INDEX_BOTTOM_RIGHT: 20,
+    INDEX_BOTTOM_LEFT: 30,
+    // Пропорции ряда следующие: в 1 ряду 2 угловые карты и 9 обычных
+    // ширина угловой карты = BASE_SIZE: 0.14, ширина обычной = MAIN_CARD_SIZE: 0.08
+    // итого 0.14 * 2 + 9 * 0.08 = 1 (где 1 это полная длина по x или y)
+    // то есть можно увеличить размер доски canvas и ничего не поедет т.к. размеры относительные
+    //размер карточки по оси y
+    BASE_SIZE: 0.14,
+    //размер карточки по оси x
+    MAIN_CARD_SIZE: 0.08,
+    //Координаты расположения фишки на карточке относительно карточки (0 - начало, 1 - конец)
+    //размер карточки по оси y
+    CHIP_SHIFT: 0.25,
+    //размер карточки по оси x
+    CHIP_CENTER: 0.5,
+  }
+  //Базовый размер (для вертикальных карточек - высота, для горизонтальных - ширина)
+  readonly baseSize: number
+  //Позиционирование карточки
+  readonly orientation: OrientationEnum
+  readonly position: PositionEnum
+  readonly type: CardTypeEmum
+  //позиция для размещения фишек
+  chipPosition = {
+    x: 0,
+    y: 0,
+  }
+
+  private backgroundColor: string = ThemeConfig.card.normalColor
+
+  protected constructor({ index, canvasSize, ctx, type }: ICardRect) {
+    super({ ctx })
+    this.type = type
+    this.orientation = CardRect.getOrientation(index)
+    this.position = CardRect.getPosition(index)
+    this.baseSize = Math.floor(canvasSize * CardRect.CONST.BASE_SIZE)
+  }
+
+  protected get propsForElements() {
+    return {
+      ...this.sizeAndCtx,
+      position: this.position,
+      orientation: this.orientation,
+    }
+  }
+
+  protected setSizeCard(index: number, canvasSize: number) {
+    const size = this.getSizesCard(index, canvasSize)
+    super.setSize(size)
+    this.chipPosition = this.getChipPosition(index)
+  }
+
+  private static getOrientation(index: number): OrientationEnum {
+    if (index % CardRect.CONST.INDEX_CORNER === 0) {
+      //при индексе кратном десяти - угловая карточка
+      return OrientationEnum.Corner
+    }
+
+    const isTop = index < CardRect.CONST.INDEX_TOP_RIGHT
+    const isBottom =
+      index > CardRect.CONST.INDEX_BOTTOM_RIGHT &&
+      index < CardRect.CONST.INDEX_BOTTOM_LEFT
+    if (isTop || isBottom) {
```

Comment:
[Предложение]
Тут проверяется position карточки. Можно получить значение  из функции getPosition
```suggestion
    const position = CardRect.getPosition(index);
    if (position === PositionEnum.Top || position === PositionEnum.Bottom) {
```